types/ts3.9.4/index.d.ts

```diff
@@ -23,6 +24,11 @@ import { DefaultParams, Params, Match, MatcherFn } from "../matcher";
 export * from "../matcher";
 export * from "../use-location";
 
+// React <18 only: fixes incorrect `ReactNode` declaration that had `{}` in the union.
+// This issue has been fixed in React 18 type declaration.
+// https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210
+type ReactNode = ReactChild | ReactNode[] | ReactPortal | boolean | null | undefined;
```

Comment:
Looks good. Only other thing I will suggest is to change
`ReactNode[]` to `Iterable<ReactNode>`. This was a change in DT that I overlooked. See details here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56623

Also worth noting, react has allowed Iterables prior to 18, it was always just incorrectly typed in DT (a problem which was masked by the inclusion of `{}`). https://github.com/facebook/react/blob/v17.0.2/packages/shared/ReactTypes.js#L20

Reply:
Good one! Fixed in https://github.com/molefrog/wouter/pull/263/commits/a3c480e87a9e3e831a726765853fe130c1241021